src/persist-client/src/usage.rs

```diff
@@ -82,24 +86,47 @@ mod tests {
             .await;
         write.expect_append(&data[..1], vec![0], vec![2]).await;
 
-        // write two rows into shard 2
+        // write two rows into shard 2 from writer 1
         let (mut write, _) = client
             .expect_open::<String, String, u64, i64>(shard_id_two)
             .await;
-        write.expect_append(&data[1..], vec![0], vec![4]).await;
+        write.expect_append(&data[1..3], vec![0], vec![4]).await;
+        let writer_one = write.writer_id.clone();
+
+        // write one row into shard 2 from writer 2
+        let (mut write, _) = client
+            .expect_open::<String, String, u64, i64>(shard_id_two)
+            .await;
+        write.expect_append(&data[4..], vec![0], vec![5]).await;
+        let writer_two = write.writer_id.clone();
 
         let usage = StorageUsageClient::open_from_blob(Arc::clone(&write.blob));
+
         let shard_one_size = usage
-            .shard_size(&shard_id_one)
+            .size(BlobKeyPrefix::Shard(&shard_id_one))
             .await
             .expect("must have shard size");
         let shard_two_size = usage
-            .shard_size(&shard_id_two)
+            .size(BlobKeyPrefix::Shard(&shard_id_two))
+            .await
+            .expect("must have shard size");
+        let writer_one_size = usage
+            .size(BlobKeyPrefix::Writer(&shard_id_two, &writer_one))
+            .await
+            .expect("must have shard size");
+        let writer_two_size = usage
+            .size(BlobKeyPrefix::Writer(&shard_id_two, &writer_two))
+            .await
+            .expect("must have shard size");
+        let all_size = usage
+            .size(BlobKeyPrefix::All)
             .await
             .expect("must have shard size");
 
         assert!(shard_one_size > 0);
```

Comment:
quite optional: I have a mild preference for asserting the actual size of each of these. it's a touch more obvious while scanning and I don't think it'll be much of a maintenance burden (famous last words)

Reply:
ah yeah I wasn't sure on this, since the size calc isn't as simple as just asserting the # of bytes in the tuple  we wrote. I can codify the values it's returning right now, though that sketches me out a little bit since it's kind of arbitrary

Reply:
yup, that's a very reasonable counter-argument and exactly why I made this super explicitly "quite optional". I'm gonna let you make the judgement call on this one :)